src/librustc_passes/stability.rs

```diff
@@ -55,12 +55,20 @@ impl<'a, 'tcx> Annotator<'a, 'tcx> {
         attrs: &[Attribute],
         item_sp: Span,
         kind: AnnotationKind,
+        inherit_deprecation: bool,
```

Comment:
Remind me what happens if we inherit deprecation for generic parameters?

Reply:
The reason was similar it, likely is no longer needed, now that `init` is gone.

Reply:
Hmm, I don't like it. This is going to be very annoying next time a type or trait with a generic parameter is deprecated. It will be even worse if there are multiple type parameters. I would like to revert this commit.

```
//~^ use of deprecated item 'unstable_generic_param::Struct5': test [deprecated]
 //~^^ use of deprecated item 'unstable_generic_param::Struct5::A': test [deprecated]
```

Reply:
I was wrong this commit is untenable, while my local build succeeded CI did not.

Reply:
I suppose we don't usually have nested items that individually support stability attributes, so this problem usually doesn't occur. Okay, I'm happy passing this around as a flag, though perhaps an enum would be better:
```rust
enum InheritDeprecation {
	Yes,
	No,
}
```
Then you can add a comment to `InheritDeprecation` explaining why this is necessary.